src/Module/Api/Mastodon/Timelines/PublicTimeline.php

```diff
@@ -42,40 +43,55 @@ public static function rawContent(array $parameters = [])
 		// Show only local statuses? Defaults to false.
 		$local = (bool)!isset($_REQUEST['local']) ? false : ($_REQUEST['local'] == 'true');
 		// Show only remote statuses? Defaults to false.
-		$remote = (bool)!isset($_REQUEST['remote']) ? false : ($_REQUEST['remote'] == 'true'); // Currently not supported
+		$remote = (bool)!isset($_REQUEST['remote']) ? false : ($_REQUEST['remote'] == 'true');
 		// Show only statuses with media attached? Defaults to false.
 		$only_media = (bool)!isset($_REQUEST['only_media']) ? false : ($_REQUEST['only_media'] == 'true'); // Currently not supported
 		// Return results older than this id
 		$max_id = (int)!isset($_REQUEST['max_id']) ? 0 : $_REQUEST['max_id'];
 		// Return results newer than this id
 		$since_id = (int)!isset($_REQUEST['since_id']) ? 0 : $_REQUEST['since_id'];
 		// Return results immediately newer than this id
-		$min_id = (int)!isset($_REQUEST['min_id']) ? 0 : $_REQUEST['min_id']; // Currently not supported
+		$min_id = (int)!isset($_REQUEST['min_id']) ? 0 : $_REQUEST['min_id'];
 		// Maximum number of results to return. Defaults to 20.
 		$limit = (int)!isset($_REQUEST['limit']) ? 20 : $_REQUEST['limit'];
 
-		$condition = ['gravity' => [GRAVITY_PARENT, GRAVITY_COMMENT], 'private' => Item::PUBLIC];
+		$params = ['order' => ['uri-id' => true], 'limit' => $limit];
+
+		$condition = ['gravity' => [GRAVITY_PARENT, GRAVITY_COMMENT], 'private' => Item::PUBLIC, 'network' => Protocol::FEDERATED];
 		if ($local) {
 			$condition['origin'] = true;
 		} else {
 			$condition['uid'] = 0;
 		}
 
+		if ($remote) {
+			$condition = DBA::mergeConditions($condition, ["NOT `uri-id` IN (SELECT `uri-id` FROM `item` WHERE `origin`)"]);
```

Comment:
Isn't this the same as `` NOT `origin` `` or `` `origin` IS NULL `` or `` `origin` = '' `` without the subquery?

Reply:
No. We query posts with "uid=0" here - and they never have "origin" set. Only the personal post from the original author has set this value.

Reply:
Ah, multiple `item` rows with the same `uri-id`, got it.